### ISSUE TITLE
💅 False positive in `useConst`

### ISSUE THREAD

**Issue by ematipico** [biomejs/biome#2575]
### Environment information

```bash
CLI:
  Version:                      1.7.1
  Color support:                true

Platform:
  CPU Architecture:             aarch64
  OS:                           macos

Environment:
  BIOME_LOG_DIR:                unset
  NO_COLOR:                     unset
  TERM:                         "xterm-256color"
  JS_RUNTIME_VERSION:           "v20.11.1"
  JS_RUNTIME_NAME:              "node"
  NODE_PACKAGE_MANAGER:         "pnpm/8.6.12"

Biome Configuration:
  Status:                       Loaded successfully
  Formatter disabled:           false
  Linter disabled:              false
  Organize imports disabled:    false
  VCS disabled:                 true

Linter:
  Recommended:                  false
  All:                          false
  Rules:                        complexity/noBannedTypes = "error"
                                complexity/noUselessTypeConstraint = "error"
                                complexity/useLiteralKeys = "error"
                                complexity/useOptionalChain = "error"
                                correctness/noPrecisionLoss = "error"
                                correctness/noUnusedVariables = "error"
                                style/noInferrableTypes = "error"
                                style/noNamespace = "error"
                                style/useAsConstAssertion = "error"
                                style/useBlockStatements = "off"
                                style/useConsistentArrayType = "error"
                                style/useConst = "off"
                                style/useForOf = "error"
                                style/useShorthandFunctionType = "error"
                                suspicious/noEmptyBlockStatements = "off"
                                suspicious/noExplicitAny = "off"
                                suspicious/noExtraNonNullAssertion = "error"
                                suspicious/noMisleadingInstantiator = "error"
                                suspicious/noUnsafeDeclarationMerging = "error"
                                suspicious/useAwait = "off"
                                suspicious/useNamespaceKeyword = "error"

Workspace:
  Open Documents:               0
```

### Rule name

lint/style/useConst

### Playground link

https://biomejs.dev/playground/?code=ZQB4AHAAbwByAHQAIABhAHMAeQBuAGMAIABmAHUAbgBjAHQAaQBvAG4AIABmACgAKQAgAHsACgAJAGwAZQB0ACAAYQA7AAoACQBhACAAPQAgAGEAdwBhAGkAdAAgAGYAZgAoACkAOwAKAH0A

### Expected result

It should not trigger because the variable is actually assigned.

### Code of Conduct

- [X] I agree to follow Biome's Code of Conduct

**Comment by Conaclos** [comment]
I think it is intended because the variable is not initialized and assigned a single time.
If I remember correctly, the rule checks that the declaration and the single assignment are in the same scope, otherwise, this is ignored.

**Comment by ematipico** [comment]
If what you say is correct, then we should:
- document the behaviour;
- provide a better diagnostic
